### PR TITLE
Misc: fix rubocop, update deps, test more ruby versions

### LINKED ISSUE #49: Ruby 3.1 support
Is there a reason why Ruby 3.1 is not supported?

https://github.com/salemove/jaeger-client-ruby/blob/9a86e9739dbda5cc8f4952c64043e1fec91c9512/jaeger-client.gemspec#L23

p.s. the CI is currently red

### PR DESCRIPTION
**Pull request by SeanLF**
- run rubocop autocorrect (needed to ignore one warning as it breaks the tests)
- test against ruby `3.1` 👉 fixes #49 
- run CI on the 1st of every month
- bump dependencies
- add MFA required on rubygems (necessary in order to release new versions)

### CODE REVIEW

**Comment by SeanLF** [comment]
@indrekj could you please approve the running of the workflow?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,16 +2,17 @@ name: Ruby
 
 on:
   push:
-    branches: [ master ]
   pull_request:
     branches: [ master ]
+  schedule:
+    - cron: '0 0 1 * *'
```
Comment (indrekj) [inline .github/workflows/ci.yml:8]: can you add a comment above this to describe it in human?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,16 +2,17 @@ name: Ruby
...
 
 jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby-version: ['2.7.2', '3.0']
+        ruby-version: ['2.7.2', '3.0', '3.1', 'head']
```
Comment (indrekj) [inline .github/workflows/ci.yml:15]: As the gemspec already has upper boundary of `< 3.2` I don't think we should include `head` here

**Review by indrekj — APPROVED** [review]

**Comment by SeanLF** [comment]
@indrekj Could you please bump the version of the gem and release it on rubygems?

**Comment by indrekj** [comment]
released as 1.3.0

**Comment by SeanLF** [comment]
Thanks!

**Pull request merged**